### ISSUE TITLE
Adding projector fails with Python error while in Edit mode

### ISSUE THREAD

**Issue by drlight-code** [Ocupe/Projectors#24]
```
Python: Traceback (most recent call last):
  File "/home/bzk/.config/blender/3.6/scripts/addons/Projectors-master/projector.py", line 523, in execute
    projector = create_projector(context)
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bzk/.config/blender/3.6/scripts/addons/Projectors-master/projector.py", line 467, in create_projector
    bpy.ops.object.light_add(type='SPOT', location=(0, 0, 0))
  File "/usr/share/blender/3.6/scripts/modules/bpy/ops.py", line 113, in __call__
    ret = _op_call(self.idname_py(), None, kw)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Operator bpy.ops.object.light_add.poll() failed, context is incorrect
```

**Comment by Ocupe** [comment]
Thank you @drlight-code for the bug report. Probably related to the new version of Blender. I'll try to fix it this weekend. Until then you have to stick with an older version of Blender. :)

**Comment by drlight-code** [comment]
Hi, working around by toggling out of edit mode first is not a problem at all, I just wanted to report this!

**Comment by Ocupe** [comment]
Should be fixed/prevented with #31
